### ISSUE TITLE
Issues with Pyth Oracle Contract Update to Beta5 on Updated Toolchain

### ISSUE THREAD

**Issue by chlenc** [FuelLabs/Pyth-integration#30]
# Issues with Pyth Oracle Contract Update to Beta5 on Updated Toolchain

Hello Fuel community,

I've recently undertaken the task of updating my Spark contracts to the beta5 version and, as part of this process, needed a functional Pyth oracle. To achieve this, I cloned the Pyth oracle repository and updated the fuels-rs dependency to version 0.54 (because I need beta5 support). After that, I rebuilt the contract using the latest toolchain. 
pr: https://github.com/FuelLabs/Pyth-integration/pull/29

However, since the update, I'm encountering an issue during testing. The tests that previously passed without any problems are now failing with the following error:

```
test functions::pyth_core::ema_price::success::gets_ema_price ... FAILED
test functions::pyth_core::price::success::gets_price ... FAILED
test functions::pyth_core::parse_price_feed_updates::success::parses_price_feed_updates ... FAILED
test functions::pyth_core::update_fee::success::gets_update_fee ... FAILED
test functions::pyth_core::price_no_older_than::success::gets_price_no_older_than ... FAILED
test functions::pyth_core::ema_price_unsafe::success::gets_ema_price_unsafe ... FAILED
test functions::pyth_core::price_unsafe::success::gets_price_unsafe ... FAILED
test functions::pyth_core::ema_price_no_older_than::success::gets_ema_price_no_older_than ... FAILED
test functions::pyth_core::update_price_feeds::success::updates_price_feeds ... FAILED
test functions::pyth_info::price_feed_unsafe::success::gets_price_feed ... FAILED
test functions::pyth_core::update_price_feeds_if_necessary::success::updates_price_feeds_if_necessary ... FAILED
test functions::pyth_init::constuctor::success::constructs ... FAILED

failures:

---- functions::pyth_core::ema_price::success::gets_ema_price stdout ----
thread 'functions::pyth_core::ema_price::success::gets_ema_price' panicked at pyth-contract/tests/utils/interface/pyth_init.rs:24:10:
called `Result::unwrap()` on an `Err` value: RevertTransactionError { reason: "InvalidMetadataIdentifier", revert_id: 0, receipts: [Call { id: 0000000000000000000000000000000000000000000000000000000000000000, to: acc609aa561bc43f4ee7e50337152b0701f3553cdd39e3942fedcff23b21e685, amount: 0, asset_id: 0000000000000000000000000000000000000000000000000000000000000000, gas: 447, param1: 2147988303, param2: 10448, pc: 13232, is: 13232 }, Panic { id: acc609aa561bc43f4ee7e50337152b0701f3553cdd39e3942fedcff23b21e685, reason: PanicInstruction { reason: InvalidMetadataIdentifier, instruction: GTF { dst: 0x10, arg: 0x12, selector: 257 } (bytes: 61 41 21 01) }, pc: 222500, is: 13232, contract_id: None }, ScriptResult { result: Panic, gas_used: 1831 }] }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- functions::pyth_core::price::success::gets_price stdout ----
thread 'functions::pyth_core::price::success::gets_price' panicked at pyth-contract/tests/utils/interface/pyth_init.rs:24:10:
called `Result::unwrap()` on an `Err` value: RevertTransactionError { reason: "InvalidMetadataIdentifier", revert_id: 0, receipts: [Call { id: 0000000000000000000000000000000000000000000000000000000000000000, to: acc609aa561bc43f4ee7e50337152b0701f3553cdd39e3942fedcff23b21e685, amount: 0, asset_id: 0000000000000000000000000000000000000000000000000000000000000000, gas: 447, param1: 2147988303, param2: 10448, pc: 13232, is: 13232 }, Panic { id: acc609aa561bc43f4ee7e50337152b0701f3553cdd39e3942fedcff23b21e685, reason: PanicInstruction { reason: InvalidMetadataIdentifier, instruction: GTF { dst: 0x10, arg: 0x12, selector: 257 } (bytes: 61 41 21 01) }, pc: 222500, is: 13232, contract_id: None }, ScriptResult { result: Panic, gas_used: 1831 }] }

---- functions::pyth_core::parse_price_feed_updates::success::parses_price_feed_updates stdout ----
thread 'functions::pyth_core::parse_price_feed_updates::success::parses_price_feed_updates' panicked at pyth-contract/tests/utils/interface/pyth_init.rs:24:10:
called `Result::unwrap()` on an `Err` value: RevertTransactionError { reason: "InvalidMetadataIdentifier", revert_id: 0, receipts: [Call { id: 0000000000000000000000000000000000000000000000000000000000000000, to: acc609aa561bc43f4ee7e50337152b0701f3553cdd39e3942fedcff23b21e685, amount: 0, asset_id: 0000000000000000000000000000000000000000000000000000000000000000, gas: 447, param1: 2147988303, param2: 10448, pc: 13232, is: 13232 }, Panic { id: acc609aa561bc43f4ee7e50337152b0701f3553cdd39e3942fedcff23b21e685, reason: PanicInstruction { reason: InvalidMetadataIdentifier, instruction: GTF { dst: 0x10, arg: 0x12, selector: 257 } (bytes: 61 41 21 01) }, pc: 222500, is: 13232, contract_id: None }, ScriptResult { result: Panic, gas_used: 1831 }] }

---- functions::pyth_core::update_fee::success::gets_update_fee stdout ----
thread 'functions::pyth_core::update_fee::success::gets_update_fee' panicked at pyth-contract/tests/utils/interface/pyth_init.rs:24:10:
called `Result::unwrap()` on an `Err` value: RevertTransactionError { reason: "InvalidMetadataIdentifier", revert_id: 0, receipts: [Call { id: 0000000000000000000000000000000000000000000000000000000000000000, to: acc609aa561bc43f4ee7e50337152b0701f3553cdd39e3942fedcff23b21e685, amount: 0, asset_id: 0000000000000000000000000000000000000000000000000000000000000000, gas: 447, param1: 2147988303, param2: 10448, pc: 13232, is: 13232 }, Panic { id: acc609aa561bc43f4ee7e50337152b0701f3553cdd39e3942fedcff23b21e685, reason: PanicInstruction { reason: InvalidMetadataIdentifier, instruction: GTF { dst: 0x10, arg: 0x12, selector: 257 } (bytes: 61 41 21 01) }, pc: 222500, is: 13232, contract_id: None }, ScriptResult { result: Panic, gas_used: 1831 }] }

---- functions::pyth_core::price_no_older_than::success::gets_price_no_older_than stdout ----
thread 'functions::pyth_core::price_no_older_than::success::gets_price_no_older_than' panicked at pyth-contract/tests/utils/interface/pyth_init.rs:24:10:
called `Result::unwrap()` on an `Err` value: RevertTransactionError { reason: "InvalidMetadataIdentifier", revert_id: 0, receipts: [Call { id: 0000000000000000000000000000000000000000000000000000000000000000, to: acc609aa561bc43f4ee7e50337152b0701f3553cdd39e3942fedcff23b21e685, amount: 0, asset_id: 0000000000000000000000000000000000000000000000000000000000000000, gas: 447, param1: 2147988303, param2: 10448, pc: 13232, is: 13232 }, Panic { id: acc609aa561bc43f4ee7e50337152b0701f3553cdd39e3942fedcff23b21e685, reason: PanicInstruction { reason: InvalidMetadataIdentifier, instruction: GTF { dst: 0x10, arg: 0x12, selector: 257 } (bytes: 61 41 21 01) }, pc: 222500, is: 13232, contract_id: None }, ScriptResult { result: Panic, gas_used: 1831 }] }

---- functions::pyth_core::ema_price_unsafe::success::gets_ema_price_unsafe stdout ----
thread 'functions::pyth_core::ema_price_unsafe::success::gets_ema_price_unsafe' panicked at pyth-contract/tests/utils/interface/pyth_init.rs:24:10:
called `Result::unwrap()` on an `Err` value: RevertTransactionError { reason: "InvalidMetadataIdentifier", revert_id: 0, receipts: [Call { id: 0000000000000000000000000000000000000000000000000000000000000000, to: acc609aa561bc43f4ee7e50337152b0701f3553cdd39e3942fedcff23b21e685, amount: 0, asset_id: 0000000000000000000000000000000000000000000000000000000000000000, gas: 447, param1: 2147988303, param2: 10448, pc: 13232, is: 13232 }, Panic { id: acc609aa561bc43f4ee7e50337152b0701f3553cdd39e3942fedcff23b21e685, reason: PanicInstruction { reason: InvalidMetadataIdentifier, instruction: GTF { dst: 0x10, arg: 0x12, selector: 257 } (bytes: 61 41 21 01) }, pc: 222500, is: 13232, contract_id: None }, ScriptResult { result: Panic, gas_used: 1831 }] }

---- functions::pyth_core::price_unsafe::success::gets_price_unsafe stdout ----
thread 'functions::pyth_core::price_unsafe::success::gets_price_unsafe' panicked at pyth-contract/tests/utils/interface/pyth_init.rs:24:10:
called `Result::unwrap()` on an `Err` value: RevertTransactionError { reason: "InvalidMetadataIdentifier", revert_id: 0, receipts: [Call { id: 0000000000000000000000000000000000000000000000000000000000000000, to: acc609aa561bc43f4ee7e50337152b0701f3553cdd39e3942fedcff23b21e685, amount: 0, asset_id: 0000000000000000000000000000000000000000000000000000000000000000, gas: 447, param1: 2147988303, param2: 10448, pc: 13232, is: 13232 }, Panic { id: acc609aa561bc43f4ee7e50337152b0701f3553cdd39e3942fedcff23b21e685, reason: PanicInstruction { reason: InvalidMetadataIdentifier, instruction: GTF { dst: 0x10, arg: 0x12, selector: 257 } (bytes: 61 41 21 01) }, pc: 222500, is: 13232, contract_id: None }, ScriptResult { result: Panic, gas_used: 1831 }] }

---- functions::pyth_core::ema_price_no_older_than::success::gets_ema_price_no_older_than stdout ----
thread 'functions::pyth_core::ema_price_no_older_than::success::gets_ema_price_no_older_than' panicked at pyth-contract/tests/utils/interface/pyth_init.rs:24:10:
called `Result::unwrap()` on an `Err` value: RevertTransactionError { reason: "InvalidMetadataIdentifier", revert_id: 0, receipts: [Call { id: 0000000000000000000000000000000000000000000000000000000000000000, to: acc609aa561bc43f4ee7e50337152b0701f3553cdd39e3942fedcff23b21e685, amount: 0, asset_id: 0000000000000000000000000000000000000000000000000000000000000000, gas: 447, param1: 2147988303, param2: 10448, pc: 13232, is: 13232 }, Panic { id: acc609aa561bc43f4ee7e50337152b0701f3553cdd39e3942fedcff23b21e685, reason: PanicInstruction { reason: InvalidMetadataIdentifier, instruction: GTF { dst: 0x10, arg: 0x12, selector: 257 } (bytes: 61 41 21 01) }, pc: 222500, is: 13232, contract_id: None }, ScriptResult { result: Panic, gas_used: 1831 }] }

---- functions::pyth_core::update_price_feeds::success::updates_price_feeds stdout ----
thread 'functions::pyth_core::update_price_feeds::success::updates_price_feeds' panicked at pyth-contract/tests/utils/interface/pyth_init.rs:24:10:
called `Result::unwrap()` on an `Err` value: RevertTransactionError { reason: "InvalidMetadataIdentifier", revert_id: 0, receipts: [Call { id: 0000000000000000000000000000000000000000000000000000000000000000, to: acc609aa561bc43f4ee7e50337152b0701f3553cdd39e3942fedcff23b21e685, amount: 0, asset_id: 0000000000000000000000000000000000000000000000000000000000000000, gas: 447, param1: 2147988303, param2: 10448, pc: 13232, is: 13232 }, Panic { id: acc609aa561bc43f4ee7e50337152b0701f3553cdd39e3942fedcff23b21e685, reason: PanicInstruction { reason: InvalidMetadataIdentifier, instruction: GTF { dst: 0x10, arg: 0x12, selector: 257 } (bytes: 61 41 21 01) }, pc: 222500, is: 13232, contract_id: None }, ScriptResult { result: Panic, gas_used: 1831 }] }

---- functions::pyth_info::price_feed_unsafe::success::gets_price_feed stdout ----
thread 'functions::pyth_info::price_feed_unsafe::success::gets_price_feed' panicked at pyth-contract/tests/utils/interface/pyth_init.rs:24:10:
called `Result::unwrap()` on an `Err` value: RevertTransactionError { reason: "InvalidMetadataIdentifier", revert_id: 0, receipts: [Call { id: 0000000000000000000000000000000000000000000000000000000000000000, to: acc609aa561bc43f4ee7e50337152b0701f3553cdd39e3942fedcff23b21e685, amount: 0, asset_id: 0000000000000000000000000000000000000000000000000000000000000000, gas: 447, param1: 2147988303, param2: 10448, pc: 13232, is: 13232 }, Panic { id: acc609aa561bc43f4ee7e50337152b0701f3553cdd39e3942fedcff23b21e685, reason: PanicInstruction { reason: InvalidMetadataIdentifier, instruction: GTF { dst: 0x10, arg: 0x12, selector: 257 } (bytes: 61 41 21 01) }, pc: 222500, is: 13232, contract_id: None }, ScriptResult { result: Panic, gas_used: 1831 }] }

---- functions::pyth_core::update_price_feeds_if_necessary::success::updates_price_feeds_if_necessary stdout ----
thread 'functions::pyth_core::update_price_feeds_if_necessary::success::updates_price_feeds_if_necessary' panicked at pyth-contract/tests/utils/interface/pyth_init.rs:24:10:
called `Result::unwrap()` on an `Err` value: RevertTransactionError { reason: "InvalidMetadataIdentifier", revert_id: 0, receipts: [Call { id: 0000000000000000000000000000000000000000000000000000000000000000, to: acc609aa561bc43f4ee7e50337152b0701f3553cdd39e3942fedcff23b21e685, amount: 0, asset_id: 0000000000000000000000000000000000000000000000000000000000000000, gas: 447, param1: 2147988303, param2: 10448, pc: 13232, is: 13232 }, Panic { id: acc609aa561bc43f4ee7e50337152b0701f3553cdd39e3942fedcff23b21e685, reason: PanicInstruction { reason: InvalidMetadataIdentifier, instruction: GTF { dst: 0x10, arg: 0x12, selector: 257 } (bytes: 61 41 21 01) }, pc: 222500, is: 13232, contract_id: None }, ScriptResult { result: Panic, gas_used: 1831 }] }

---- functions::pyth_init::constuctor::success::constructs stdout ----
thread 'functions::pyth_init::constuctor::success::constructs' panicked at pyth-contract/tests/utils/interface/pyth_init.rs:24:10:
called `Result::unwrap()` on an `Err` value: RevertTransactionError { reason: "InvalidMetadataIdentifier", revert_id: 0, receipts: [Call { id: 0000000000000000000000000000000000000000000000000000000000000000, to: acc609aa561bc43f4ee7e50337152b0701f3553cdd39e3942fedcff23b21e685, amount: 0, asset_id: 0000000000000000000000000000000000000000000000000000000000000000, gas: 445, param1: 2147988303, param2: 10448, pc: 13232, is: 13232 }, Panic { id: acc609aa561bc43f4ee7e50337152b0701f3553cdd39e3942fedcff23b21e685, reason: PanicInstruction { reason: InvalidMetadataIdentifier, instruction: GTF { dst: 0x10, arg: 0x12, selector: 257 } (bytes: 61 41 21 01) }, pc: 222500, is: 13232, contract_id: None }, ScriptResult { result: Panic, gas_used: 1791 }] }


failures:
    functions::pyth_core::ema_price::success::gets_ema_price
    functions::pyth_core::ema_price_no_older_than::success::gets_ema_price_no_older_than
    functions::pyth_core::ema_price_unsafe::success::gets_ema_price_unsafe
    functions::pyth_core::parse_price_feed_updates::success::parses_price_feed_updates
    functions::pyth_core::price::success::gets_price
    functions::pyth_core::price_no_older_than::success::gets_price_no_older_than
    functions::pyth_core::price_unsafe::success::gets_price_unsafe
    functions::pyth_core::update_fee::success::gets_update_fee
    functions::pyth_core::update_price_feeds::success::updates_price_feeds
    functions::pyth_core::update_price_feeds_if_necessary::success::updates_price_feeds_if_necessary
    functions::pyth_info::price_feed_unsafe::success::gets_price_feed
    functions::pyth_init::constuctor::success::constructs

test result: FAILED. 0 passed; 12 failed; 0 ignored; 0 measured; 0 filtered out; finished in 1.56s

error: test failed, to rerun pass `--test integration_tests`
```

For reference, here are the details of my toolchain setup:

```
Default host: aarch64-apple-darwin
fuelup home: /Users/alexey/.fuelup

installed toolchains
--------------------
latest-aarch64-apple-darwin (default)
beta-4-aarch64-apple-darwin
latest-2023-11-15-aarch64-apple-darwin (override)
my-custom-toolchain

active toolchain
-----------------
latest-2023-11-15-aarch64-apple-darwin (override), path: /Users/alexey/Desktop/Pyth-integration/fuel-toolchain.toml
  forc : 0.47.0
    - forc-client
      - forc-deploy - not found
      - forc-run - not found
    - forc-doc - not found
    - forc-explore : 0.28.1
    - forc-fmt - not found
    - forc-index : 0.23.0
    - forc-lsp - not found
    - forc-tx - not found
    - forc-wallet : 0.3.0
  fuel-core : 0.20.8
  fuel-core-keygen : Error getting version string
  fuel-indexer : 0.23.0

fuels versions
---------------
forc : 0.45
forc-wallet : 0.45
```

I'm looking for insights or suggestions on what might be causing this error and how to resolve it. Has anyone else experienced similar issues after updating to beta5 or with the latest toolchain? Any guidance or advice would be greatly appreciated.

Thank you in advance for your help!

**Comment by hummusonrails** [comment]
cc: @K1-R1

**Comment by K1-R1** [comment]
This should be resolved by updating the contract itself to beta-5. I'll open a PR to do so

**Comment by K1-R1** [comment]
@chlenc This should be resolved as of #32
